tests/transport/compliance.py

```diff
@@ -108,3 +117,15 @@ def test_connection_error(self):
         request = self.make_request()
         with pytest.raises(exceptions.TransportError):
             request(url="http://{}".format(NXDOMAIN), method="GET")
+        
+    def test_cached_request(self, server):
```

Comment:
Can we add a test for expired cached request?

Reply:
Good call. Just pushed another test that should cover this